src/features/tempRole/expireCheck.ts

```diff
@@ -34,7 +36,14 @@ export async function expireCheck(botContext: BotContext) {
       const role = guild.roles.cache.get(expiredTemporaryRole.roleId) as Role
 
       // Remove role from member
-      member.roles.remove(role)
+      try {
+        await member.roles.remove(role)
+      } catch (error) {
+        console.error(
+          `Failed to remove role "${role.name}" from "${member.user.tag}"`,
+          (error as Error).message,
+        )
+      }
     }
     // Remove expired temporary roles from database
     await prisma.tempRole.deleteMany({
```

Comment:
I think if we cannot delete role, we need to keep it in database for retry or report to mod.